CGAL_Core/include/CGAL/CORE/poly/Poly.tcc

```diff
@@ -670,19 +670,20 @@ Polynomial<NT> Polynomial<NT>::pseudoRemainder (
   contract();         // Let A = (*this).  Contract A.
   Polynomial<NT> tmpB(B);
   tmpB.contract();    // local copy of B
+  int bTrueDegree = tmpB.degree;
   C = NT(1);  // Initialized to C=1.
-  if (B.degree == -1)  {
+  if (bTrueDegree == -1)  {
     core_error("ERROR in Polynomial<NT>::pseudoRemainder :\n    -- divide by zero polynomial", __FILE__, __LINE__, false);
     return Polynomial(0);  // Unit Polynomial (arbitrary!)
   }
-  if (B.degree > degree) {
+  if (bTrueDegree > degree) {
```

Comment:
This was the specific bug. Although the apparent degree of B can be larger than the degree of A (this), its true degree might not be.
I didn't change `degree` since a call to "contract" removes the trailing zero coefficients and makes degree = true degree.